Porting/test-dist-modules.pl

```diff
@@ -0,0 +1,1087 @@
+@dists = grep { $_ ne "Safe" } @dists;
+
+for my $dist (@dists) {
+    test_dist($dist);
+}
+
+sub test_dist {
+    my ($name) = @_;
+
+    print "*** Testing $name ***\n";
+    my $dir = tempdir( CLEANUP => 1);
+    system qq(cp -a "dist/$name/." "$dir/.")
+      and die "Cannot copy files";
+    chdir $dir
+      or die "Cannot chdir $dir: $!";
+    if ($pppfile) {
+        system "cp", $pppfile, "."
+          and die "Cannot copy $pppfile to .\n";
+    }
+    if ($name eq "IO" || $name eq "threads" || $name eq "threads-shared") {
+        write_testpl();
+    }
+    unless (-f "Makefile.PL") {
+        print "  Creating Makefile.PL for $name\n";
+        my $key = "ABSTRACT_FROM";
+        my @parts = split /-/, $name;
+        my $last = $parts[-1];
+        my $fromname;
+        for my $check ("$last.pm", join("/", "lib", @parts) . ".pm") {
+            if (-f $check) {
+                $fromname = $check;
+                last;
+            }
+        }
+        $fromname
+          or die "Cannot find ABSTRACT_FROM for $name\n";
+        my $value = $fromname;
+        open my $fh, ">", "Makefile.PL"
+          or die "Cannot create Makefile.PL: $!\n";
+        # adapted from make_ext.pl
+        printf $fh <<'EOM', $name, $fromname, $key, $value;
+use strict;
+use ExtUtils::MakeMaker;
+
+# This is what the .PL extracts to. Not the ultimate file that is installed.
+# (ie Win32 runs pl2bat after this)
+
+# Doing this here avoids all sort of quoting issues that would come from
+# attempting to write out perl source with literals to generate the arrays and
+# hash.
+my @temps = 'Makefile.PL';
+foreach (glob('scripts/pod*.PL')) {
+    # The various pod*.PL extractors change directory. Doing that with relative
+    # paths in @INC breaks. It seems the lesser of two evils to copy (to avoid)
+    # the chdir doing anything, than to attempt to convert lib paths to
+    # absolute, and potentially run into problems with quoting special
+    # characters in the path to our build dir (such as spaces)
+    require File::Copy;
+
+    my $temp = $_;
+    $temp =~ s!scripts/!!;
+    File::Copy::copy($_, $temp) or die "Can't copy $temp to $_: $!";
+    push @temps, $temp;
+}
+
+my $script_ext = $^O eq 'VMS' ? '.com' : '';
+my %%pod_scripts;
+foreach (glob('pod*.PL')) {
+    my $script = $_;
+    s/.PL$/$script_ext/i;
+    $pod_scripts{$script} = $_;
+}
+my @exe_files = values %%pod_scripts;
+
+WriteMakefile(
+    NAME          => '%s',
+    VERSION_FROM  => '%s',
+    %-13s => '%s',
+    realclean     => { FILES => "@temps" },
+    (%%pod_scripts ? (
+        PL_FILES  => \%%pod_scripts,
+        EXE_FILES => \@exe_files,
+        clean     => { FILES => "@exe_files" },
+    ) : ()),
+);
+
+EOM
+        close $fh;
+    }
+    system "$^X Makefile.PL"
+      and die "$name: Makefile.PL failed\n";
+
+    my $verbose = 0;
+    system "make test TEST_VERBOSE=$verbose"
+      and die "$name: make test failed\n";
+
+    system "make install"
```

Comment:
They're installed so later tests are impacted.

This was written for use within the docker images which are setup to allow installation with the default permissions.

I can see this script being extended to allow use outside of that environment (and allowing invocation like `someperl Porting/test-dist-modules.pl Storable` rather than populating from `readdir()`) but I don't aim to do that in this PR.